File: test/conformance/host-fixtures/nullreturn_ok.js
"use strict";

// Host fixture implementation for the host-null-return-ok conformance
// test. JS null is the DEAL null sentinel for a sync function declared
// ->null — the wrapper's sync-null check accepts it and rejects any
// non-sentinel value (plain undefined included).
module.exports = {
  ping: function () {
    return null;
  },
};

File: test/conformance/host-fixtures/nullreturn_ok.java
// Host fixture implementation for the host-null-return-ok conformance
// test. Java null is the JVM null sentinel for a sync function declared
// ->null — the wrapper's sync-null check accepts it and rejects any
// non-sentinel value.
//
// The top-level class is deliberately package-private: the corpus file
// keeps its <name>.java stem while the JVM lane deploys the source
// beside the default-package artifacts under the classNameFor name
// Host<Name>.java (test/JvmConformanceTest.java HOST_JAVA pattern).
final class HostNullreturn_ok {
  public static Object ping() {
    return null;
  }
}
